[SIS-036] Add activity card page new

MinGi-K opened the pull request:
# Description

Created front end page for adding an activity card. Used React-Native-Paper package for the search bar, will replace with custom made one in future if it is not customizable enough.

Added new fonts Roboto and Mulish.

Ticket #36

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My PR title is formatted as [SIS-\<issue number\>] \<issue name\> (eg. [SIS-010] Create UI button)
- [x] I have linked my issue/ticket on the project board to this PR
- [x] I have sent my PR to #team-sistema-prs on Slack and requested two people to review
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (if required)
- [x] My changes generate no new warnings
- [x] I have tested that my feature works as expected on Android and iOS. If you are a Windows dev, please request a Mac dev to run your branch. If it's a front-end change, I have attached screenshots of what the result looks like on both platforms. 


<img width="720" alt="Screen Shot 2023-01-30 at 10 05 53 PM" src="https://user-images.githubusercontent.com/64427415/215653478-a2d304e3-1cfb-4dd3-8ae5-2107ee137e6f.png">

MinGi-K committed:
feat(editor): added TagFilter component, fonts, addActivityCard screen

MinGi-K committed:
squash

MinGi-K committed:
feat(addactivitycard): formatting and spacing

MinGi-K committed:
linked fonts to android

MinGi-K committed:
feat(addactivitycard): added searchbar to addActivityCard screen

MinGi-K committed:
merge with master

MinGi-K committed:
feat(yarn.lock and package.json): installed medi package for searchbar

MinGi-K committed:
yarn lint

MinGi-K committed:
feat(addactivitycard): changed backarrow to use react native transformer

MinGi-K committed:
git pull origin master

MinGi-K committed:
merge yarn.lock

MinGi-K committed:
why does this file say there is a merge conflict when there isn't

MinGi-K committed:
feat(addactivitycard, xcodeproj.pbxproj): ensured build works on both android and ios

added fonts to ios, made changes to info plist and added ttf files to resources. A few formatting
changes.

MinGi-K committed:
removed comma

BakerWJ commented on src/editor/screens/AddActivityCard.js:
```diff
@@ -0,0 +1,96 @@
+import TagFilter from '../components/TagFilter';
+import BackArrow from '../../../assets/backArrow.svg';
+import { Searchbar } from 'react-native-paper';
+import { mdiMagnify } from '@mdi/js';
+import { isSearchBarAvailableForCurrentPlatform } from 'react-native-screens';
```
remove unused imports

MinGi-K replied:
Done!!!

BakerWJ commented on src/editor/screens/AddActivityCard.js:
```diff
@@ -0,0 +1,96 @@
+  Text,
+  StyleSheet,
+  TouchableOpacity,
+  useState
```
useState isn't being used currently

MinGi-K replied:
It will be in the future! Currently it is set up like so:
```
const [searchQuery, setSearchQuery] = React.useState('');
```
So we can change the state of what is written into the search bar, but just no functionality is implemented yet.

BakerWJ commented on src/editor/components/TagFilter.js:
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { View, TouchableOpacity, StyleSheet, Text } from 'react-native';
```
View isn't being used currently, can be removed

MinGi-K replied:
Done!

BakerWJ commented on src/editor/screens/AddActivityCard.js:
```diff
@@ -0,0 +1,96 @@
+          onChangeText={onChangeSearch}
+          value={searchQuery}
+          icon={require('../../../assets/search.png')}
+          style={{ backgroundColor: 'white', marginTop: '3%' }}
```
consider moving to styles -- @yuemily24 should we avoid using inline styles or is it okay for small styling in individual components

Lemiex replied:
I'm not sure how accurate this is but I think emily mentioned to use SafeAreaView instead of View??

MinGi-K replied:
I can move to styles, no problem. As for @Lemiex the AddActivityCard screen is wrapped in SafeAreaView so it should be fine to have Views nested inside it.

BakerWJ commented on src/editor/screens/AddActivityCard.js:
```diff
@@ -0,0 +1,96 @@
+  header: {
+    color: 'black',
+    fontSize: 22,
+    fontFamily: 'Poppins-ExtraBold',
```
Not 100% confident, but I think that most of the designs only use Poppins or Mulish. I f you want it to be bolder consider changing font weight (I haven't looked at designs for this particular instance so ignore if I'm wrong)

Lemiex replied:
I'm so glad fonts are working now :D

Lemiex replied:
I think Baker is right. From last standup I think we mentioned we only use Poppins!

MinGi-K replied:
I changed it to only Poppins!!

BakerWJ reviewed (commented):
I haven't had the chance to try out the PR locally -- I left some comments for now and will do another review once I can compile and run the code.

MinGi-K committed:
Merge branch 'master' of https://github.com/uoftblueprint/sistema into 36-sis-036-add-activity-card-page-1-NEW

git pull origin master

MinGi-K committed:
feat(addactivitycard, tagfilter): added PR suggestions where appropriate

Changed in-line styling, removed unused imports, changed Poppins-ExtraBold font to Poppins with a
larger fontWeight.

MinGi-K committed:
squash

ramyzhang commented:
@Lemiex @BakerWJ pls review! :D

ramyzhang committed:
fix(info.plist): was getting an mui not available error (might be xcode vers)

fixed!

ramyzhang commented:
hey it's looking good! just made a quick commit bc it wasn't running on my machine, likely bc of my xcode version, and adding some stuff to Info.plist fixed it.

MinGi-K commented:
Thank you!!

yuemily24 commented:
Please fix merge conflicts and new UI changes, thanks!

MinGi-K committed:
pull origin master

MinGi-K committed:
Merge branch 'master' of https://github.com/uoftblueprint/sistema into 36-sis-036-add-activity-card-page-1-NEW

"merge origin master"

MinGi-K committed:
implemented custom search bar

MinGi-K committed:
squash

MinGi-K committed:
workspace stuff

MinGi-K committed:
feat(addactivitycard): made font size a bit smaller

yuemily24 commented on src/editor/screens/AddActivityCard.js:
```diff
@@ -0,0 +1,99 @@
+import * as React from 'react';
```
Please avoid overarching import from react and import `useState` or other functions directly.

yuemily24 commented on src/editor/screens/AddActivityCard.js:
```diff
@@ -0,0 +1,99 @@
+} from 'react-native';
+import TagFilter from '../components/TagFilter';
+import BackArrow from '../../../assets/backArrow.svg';
+// import { Searchbar } from 'react-native-paper';
```
Remove

yuemily24 commented on package.json:
```diff
@@ -26,9 +27,11 @@
     "react-native": "0.70.6",
     "react-native-fs": "^2.20.0",
     "react-native-gesture-handler": "^2.8.0",
+    "react-native-paper": "^5.1.3",
```
Remove? I believe this is the searchbar package and not needed anymore

yuemily24 commented on package.json:
```diff
@@ -26,9 +27,11 @@
     "react-native-screens": "^3.18.2",
-    "react-native-svg": "^13.6.0"
+    "react-native-svg": "^13.6.0",
+    "react-native-vector-icons": "^9.2.0"
```
What is this package for? Can it be removed? I don't see it being used anywhere. Make sure if you yarn remove to rebuild and push any build changes

yuemily24 commented on src/editor/components/Searchbar.js:
```diff
@@ -0,0 +1,58 @@
+import React from 'react';
```
Avoid big import from react

yuemily24 commented on src/editor/components/TagFilter.js:
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
```
avoid big import from react

yuemily24 commented on src/editor/screens/AddActivityCard.js:
```diff
@@ -0,0 +1,99 @@
+          placeholder="Search by title or keyword"
+          onChangeText={onChangeSearch}
+          value={searchQuery}
+          placeholderText
```
Empty? Is this diff from `placeholder` property above?

yuemily24 commented on ios/sistema/Info.plist:
```diff
@@ -85,8 +85,22 @@
-		<string>Mulish-SemiBoldItalic.ttf</string>
+		<string>Roboto-Regular.ttf</string>
+		<string>AntDesign.ttf</string>
+	  <string>Entypo.ttf</string>
+	  <string>EvilIcons.ttf</string>
+	  <string>Feather.ttf</string>
+	  <string>FontAwesome.ttf</string>
+	  <string>FontAwesome5_Brands.ttf</string>
+	  <string>FontAwesome5_Regular.ttf</string>
+	  <string>FontAwesome5_Solid.ttf</string>
+	  <string>Foundation.ttf</string>
+	  <string>Ionicons.ttf</string>
+	  <string>MaterialIcons.ttf</string>
+	  <string>MaterialCommunityIcons.ttf</string>
+	  <string>SimpleLineIcons.ttf</string>
+	  <string>Octicons.ttf</string>
+	  <string>Zocial.ttf</string>
```
I'm a bit confused why this stuff was added, since they're not being used. Were they dependencies of a package?

MinGi-K replied:
I think they came in from merge conflicts, I accepted incoming because I didn't know where they came from but just in case other people were using it.

yuemily24 replied:
Hm I've looked over most people's PRs and I don't think anyone's using this stuff. Well if you can delete them and it doesn't break anything I'd get rid of them haha

ramyzhang replied:
oh it was bc it was breaking the mui dependency and the build if these weren't in the info.plist so i added them. if no one's using the mui dependency then for sure let's remove them. mb!

yuemily24 commented on src/editor/EditorNavigator.js:
```diff
@@ -1,18 +1,24 @@
     <Stack.Navigator
       intialRouteName={STACK_SCREENS.LESSON_PLAN_EDITOR}
       screenOptions={{ headerShown: false }}>
+      <Stack.Screen
+        name={STACK_SCREENS.ADD_ACTIVITY_CARD}
+        component={AddActivityCard}
+      />
```
Please switch the order of this screen and `STACK_SCREENS.LESSON_PLAN_EDITOR`, because otherwise the stack will show what's at the top. (For some reason, `initialRouteName` doesn't kick in hm.)
```
      <Stack.Screen
        name={STACK_SCREENS.LESSON_PLAN_EDITOR}
        component={LessonPlanEditor}
      />
      <Stack.Screen
        name={STACK_SCREENS.ADD_ACTIVITY_CARD}
        component={AddActivityCard}
      />
```

yuemily24 reviewed (changes requested):
Hi Mingi, generally the code looks good, but please clean up any packages you're not using.

There are a few main features I see missing from this ticket:

- Pressing a tag should toggle it. In other words, when I press on it, it should stay selected/grey until I press on it again. My recommendation is to keep a useState to keep track of which tags are toggled. 
- Hitting the back button should bring you back to the main LP editor (look into passing in and using navigation `navigation.goBack()`)
- Inside the main LP editor, pressing the Add Activity Card button should pull up this screen. 
     - It's a `TextInput` because of Sarah's old changes, so please go in and change all of them to a static button. Should be relatively easy. 
     - The title in your Search Activity Card screen such as "**Main Lesson** Activity Card" should change based on which section you are passing in. _You don't need to fix this now_, but please fix this for your next ticket (you'll likely need it anyway when you're figuring out where to download the activity cards to RNFS and we need to make sure it's going to the right section). Just something to keep in mind. 
- Please add more margin between the title and "Tags" subtitle

![image](https://user-images.githubusercontent.com/76456096/216112095-c4792b9a-3fe0-4686-923d-dd127a0378bc.png)

MinGi-K committed:
feat: added toggability to filters

MinGi-K committed:
feat: added more margin between header and tags

MinGi-K committed:
feat: made add activity card a button, added back navigation, made header to activity screenspecific

N

MinGi-K committed:
feat(n): formatting

MinGi-K committed:
feat: implemented on change text function

MinGi-K committed:
feat: removed weird unused info.plist dependencies

MinGi-K committed:
feat: added info.plst dependencies back

N

MinGi-K committed:
squash

MinGi-K committed:
squash

ramyzhang committed:
style(addactivitycard.js): linting and styles

ramyzhang reviewed (approved):
looks good, thank you mingi!

ramyzhang committed:
Merge branch 'master' into 36-sis-036-add-activity-card-page-1-NEW

ramyzhang committed:
fix(editornavigator.js): missing comma from merge conflict reso
